src/PaymentSource.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Invoiced;
+
+class PaymentSource extends BaseObject
+{
+    /**
+     * Creates an object. This variant creates the appropriate subtype if applicable.
+     *
+     * @param array $params
+     * @param array $opts
+     *
+     * @return object newly created object
+     */
+    public function create(array $params = [], array $opts = [])
+    {
+        $response = $this->_client->request('post', $this->getEndpoint(), $params, $opts);
+
+        $obj = $response['body'];
+
+        if ($obj['object'] == 'card') {
```

Comment:
I feel like this could be handled in `Util::convertToObject` since it's repeated twice.

Reply:
It could, although from a functional perspective I think it would be less clean to make `Util::convertToObject` inspect the object it's converting. Will do either as you prefer?